src/Auth0.php

```diff
@@ -665,15 +664,17 @@ public function setAccessToken($accessToken)
      */
     public function setIdToken($idToken)
     {
-        $jwtVerifier          = new JWTVerifier([
-            'valid_audiences' => ! empty($this->idTokenAud) ? $this->idTokenAud : [ $this->clientId ],
-            'supported_algs' => $this->idTokenAlg ? [ $this->idTokenAlg ] : [ 'HS256', 'RS256' ],
-            'authorized_iss' => $this->idTokenIss ? $this->idTokenIss : [ 'https://'.$this->domain.'/' ],
-            'client_secret' => $this->clientSecret,
-            'secret_base64_encoded' => $this->clientSecretEncoded,
-            'guzzle_options' => $this->guzzleOptions,
-        ]);
-        $this->idTokenDecoded = (array) $jwtVerifier->verifyAndDecode( $idToken );
+        $sigVerifier = null;
+        if ('RS256' === $this->idTokenAlg) {
+            $jwksFetcher = new JWKFetcher($this->cacheHandler, $this->guzzleOptions);
+            $jwks        = $jwksFetcher->getKeys($this->idTokenIss.'.well-known/jwks.json');
+            $sigVerifier = new JwksVerifier( $jwks );
+        } else if ('HS256' === $this->idTokenAlg) {
+            $sigVerifier = new SymmetricVerifier($this->clientSecret);
+        }
```

Comment:
The "else" branch is being checked on the constructor. Is there any chance the user can change this on runtime and end up having an algorithm value not supported?

Reply:
> Is there any chance the user can change this on runtime and end up having an algorithm value not supported?

Not possible. But, even if it was, `IdTokenVerifier` constructor would fail.